src/robot.py

```diff
@@ -197,6 +212,10 @@ def move_robot(self):
                 observer.__call__(self.current_pose)
             # TODO remove this hardcoded drain amount with a value based on load, turning, movement, etc
             self.battery_charge.drain_battery(1.0)
+        
+        # mark that the robot has reached the shelf
+        if self.current_pose == self.shelf_pose:
```

Comment:
same question here about using end_pose instead of shelf_pose.

Reply:
I'm just trying to keep shelf_pose and goal_pose seperate to help with handling the state of the robot. Basically, if goal_pose is not actually the shelf its supposed to pick then I don't want to set has_shelf.